docs/site/Express-middleware.md

```diff
@@ -0,0 +1,229 @@
+ *
+ * @typeParam CFG - Configuration type
+ * @typeParam CTX - Context type
+ */
+export function createInterceptor<CFG, CTX extends Context = InvocationContext>(
+  middleware: string | MiddlewareFactory<CFG>,
+  middlewareConfig?: CFG,
+): GenericInterceptor<CTX> {
+  // ...
+}
+```
+
+### Adapt an Express middleware handler function to an interceptor
+
+```ts
+/**
+ * Wrap an express middleware handler function as an interceptor
+ * @param handlerFn - Express middleware handler function
+ *
+ * @typeParam CTX - Context type
+ */
+export function toInterceptor<CTX extends Context = InvocationContext>(
+  handlerFn: ExpressRequestHandler,
+): GenericInterceptor<CTX> {
+  // ...
+}
+```
+
+### Define a provider class to allow dependency injection of middleware configuration
+
+```ts
+/**
+ * Define a provider class that wraps the middleware as an interceptor
+ * @param middleware - Middleware module name or factory function
+ * @param className - Class name for the generated provider class
+ *
+ * @typeParam CFG - Configuration type
+ * @typeParam CTX - Context type
+ */
+export function defineInterceptorProvider<
+  CFG,
+  CTX extends Context = InvocationContext
+>(
+  middleware: string | MiddlewareFactory<CFG>,
+  className?: string,
+): Constructor<Provider<GenericInterceptor<CTX>>> {
+  // ...
+}
+```
+
+Alternatively, we can create a subclass of `MiddlewareInterceptorProvider`.
+
+### Use middleware as invocation interceptors
+
+With the ability to wrap Express middleware as LoopBack 4 interceptors, we can
+use the same programming model to register middleware as global interceptors or
+local interceptors denoted by `@intercept` decorators at class and method
+levels.
+
+### Introduce middleware to REST sequence of actions
+
+There are a few steps involved in the default sequence of actions. See
+`Sequence.md` for more details.
+
+It's often desirable to reuse Express middleware in the sequence to handle API
+requests/responses without reinventing the world. We now add an `Middleware`
+action as the first step in the default sequence. The action itself is an
+interceptor chain of `RequestContext`. It uses the powerful
+[extension point/extension pattern](Extension-point-and-extensions.md) to
+contribute and discover registered Express middleware.
+
+#### Register an Express middleware for the sequence
+
+````ts
+/**
+ * Bind a middleware interceptor to this server context
+ *
+ * @example
+ * ```ts
+ * server.middleware('express-middleware-name', {});
+ * ```
+ * @param middleware - Middleware module name or factory function
+ * @param middlewareConfig - Middleware config
+ * @param options - Options for registration
+ *
+ * @typeParam CFG - Configuration type
+ */
+ middleware<CFG>(
+   middleware: string | MiddlewareFactory<CFG>,
+   middlewareConfig?: CFG,
+   options: MiddlewareActionBindingOptions = {},
+ ): Binding<RequestInterceptor> {
+   // ...
+ }
+````
+
+```ts
+const binding = app.middleware(spy, spyConfig, {
```

Comment:
Would `app.middleware(mw, mwConfig, {` or `app.middleware(middleware, middleware Config, {` be more descriptive?

Reply:
Or include the definition of `spy` and `spyConfig` in the code.